scripts/sync_makefiles.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# GITHUB_TOKEN required scope: repo.repo_public
+
+git_mail="prombot@gmail.com"
+git_user="prombot"
+branch="makefile_common"
+commit_msg="makefile: update Makefile.common with newer version"
+pr_title="Synchronize Makefile.common from prometheus/prometheus"
+pr_msg="Propagating changes from master Makefile.common located in prometheus/prometheus."
+org="prometheus"
+target_repos=("alertmanager"
+	"blackbox_exporter"
+	"client_golang"
+	"common"
+	"consul_exporter"
+	"graphite_exporter"
+	"haproxy_exporter"
+	"influxdb_exporter"
+	"memcached_exporter"
+	"mysqld_exporter"
+	"node_exporter"
+	"procfs"
+	"prom2json"
+	"promu"
```

Comment:
There are more repositories to include. This should do the trick: `curl -s https://api.github.com/orgs/prometheus/repos?type=sources | jq -r '.[]  |.name'`

Reply:
You're right I missed some, but not all of repositories in prometheus org have `Makefile.common` in top level directory or even present anywhere.

I am using this to detect which should be included:
```sh
#!/bin/bash
for repo in $(curl -s https://api.github.com/orgs/prometheus/repos?type=sources | jq -r '.[].name'); do
    curl -s --fail https://raw.githubusercontent.com/prometheus/$repo/master/Makefile.common >/dev/null && echo "$repo"
done
```

Initial design for this was actually looping over all repositories in target organization using similar `curl` statement, but after seeing that a number of exclusions is more or less equal to inclusions I decided to remove that one API GitHub call.

Reply:
Nice, might want to filter "prometheus" from the list. `jq -r '.[] | select( .name != "prometheus" ) | .name'`

Reply:
Seems like I missed those:
```
	"pushgateway"
	"snmp_exporter"
	"statsd_exporter"
```

:)

Reply:
I figure it's better to `curl --fail` below when checksumming the targets.

Reply:
I'd prefer to get the repositories from GitHub every time the script is executed rather than hardcode the list. Then you can skip repositories without `Makefile.common` in the for loop. The rationale is that we may have more repositories in the future or an existing repository might start using `Makefile.common`. Although these things change rarely, I'd prefer not to miss them.

Reply:
:+1: This reminded me that the `collectd_exporter` was missing the upgrade to Makefile.common, so now that's in the list. :smile: